tests/Functional/EventsTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/*
+ * This file is part of the WouterJEloquentBundle package.
+ *
+ * (c) 2017 Wouter de Jong
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace WouterJ\EloquentBundle\Functional;
+
+use AppBundle\Model\User;
+use AppBundle\Model\SoftDeleteUser;
+use Illuminate\Database\Schema\Blueprint;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use WouterJ\EloquentBundle\Facade\Schema;
+
+/**
+ * @author Wouter J <wouter@wouterj.nl>
+ */
+class EventsTest extends KernelTestCase
+{
+    protected static function getKernelClass()
+    {
+        require_once __DIR__.'/app/TestKernel.php';
+
+        return 'TestKernel';
+    }
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        static::bootKernel();
+
+        if (!Schema::hasTable('users')) {
+            Schema::create('users', function (Blueprint $table) {
+                $table->increments('id');
+                $table->string('name');
+                $table->string('email');
+                $table->string('password');
+                $table->timestamps();
+            });
+        }
+    }
+
+    /** @dataProvider getEventsData */
+    public function testEvents($setup, $execute, array $expectedEvents)
+    {
+        ob_start();
```

Comment:
Nope, just back from a holiday and my brain isn't thinking 100% straight yet ;) Used a better technique now.